src/computation/logic/distribute-levelling-seats.ts

```diff
@@ -90,6 +65,164 @@ export function distributeLevelingSeats(
         );
     }
 
-    const levelingSeatDistribution = dictionaryToArray(partyRestQuotients);
-    return levelingSeatDistribution;
+    const levelingSeatDistributionArray = dictionaryToArray(partyRestQuotients);
+    return levelingSeatDistributionArray;
+}
+
+/**
+ * Filters out parties that do not receive more seats in the national distribution than the district distribution.
+ * It returns the remaining party codes and the final national distribution of seats.
+ *
+ * @param levelingPartyCodes The party codes to filter
+ * @param payload The computation payload
+ * @param partyResults The party results for each party
+ */
+function nationalDistributionFilter(
+    levelingPartyCodes: string[],
+    payload: ComputationPayload,
+    partyResults: Dictionary<PartyResult>
+): { nationalDistribution: DistributionResult; levelingPartyCodes: string[] } {
+    let localLevelingPartyCodes = [...levelingPartyCodes];
+    let nationalDistribution: DistributionResult;
+
+    let unfilteredPartiesLength;
+    let filteredPartiesLength = localLevelingPartyCodes.length;
+    do {
```

Comment:
I am very skeptical to the usage of do-whiles, they are generally just added complexity.

Reply:
Made the function recursive instead.